pkgs/top-level/all-packages.nix

```diff
@@ -14916,8 +14916,6 @@ with pkgs;
 
   automake116x = callPackage ../development/tools/misc/automake/automake-1.16.x.nix { };
 
-  automoc4 = callPackage ../development/tools/misc/automoc4 { };
-
```

Comment:
Don't forget to add a throw message at `top-level/aliases.nix`.

Reply:
why? honest question. do you think anyone would benefit from that?

Reply:
Everyone benefits from this.

Sometimes people create flakes and overlays around Nixpkgs in order to create and modify its packages. 
If by any reason someone uses a package or expression dependent on `automoc4`, the throw will tell them that automoc was removed from Nixpkgs, and the person can act accordingly.

And the same applies if someone tries to put this package back on Nixpkgs, whether directly or as a dependency of some obscure package (sometimes inadvertently - as it happened with myself).

Reply:
Aha. They'll get an exception either way. They'll know `automoc4` was removed, because what else should have happened to it?

The only thing adding an alias does is making `aliases.nix` longer.

Reply:
> Sometimes people create flakes and overlays around Nixpkgs

I co-created projects like https://github.com/helsinki-systems/bbb4nix do you really think I don't know this?

> And the same applies if someone tries to put this package back on Nixpkgs, whether directly or as a dependency of some obscure package (sometimes inadvertently - as it happened with myself).

I'd veto that and enforce that, if necessary, if it comes to this package. It's old and long dead and depends on qt4, which has been EOL since the end of 2015.

Reply:
> I co-created projects like https://github.com/helsinki-systems/bbb4nix do you really think I don't know this?

Are you angry because I gave an example?

Are you seriously infuriated because I gave an easy and understandable motivation for my advice (that is not a set-on-stone thing, just to be clear)?

> I'd veto that and enforce that, if necessary, if it comes to this package. It's old and long dead and depends on qt4, which has been EOL since the end of 2015.

Python 2 and lilyterm [gave me some nightmares](https://github.com/NixOS/nixpkgs/pull/157959).